src/Microsoft.ML.Experimental/DataLoadSave/Database/DatabaseLoaderCursor.cs

```diff
@@ -236,79 +236,79 @@ private Delegate CreateGetterDelegate<TValue>(int col)
             private ValueGetter<bool> CreateBooleanGetterDelegate(ColInfo colInfo)
             {
                 int columnIndex = GetColumnIndex(colInfo);
-                return (ref bool value) => value = DataReader.GetBoolean(columnIndex);
+                return (ref bool value) => value = DataReader.IsDBNull(columnIndex) ? default : DataReader.GetBoolean(columnIndex);
```

Comment:
I don't think blindly turning `null` into `default` is really the correct thing to do here.

I wonder if we should have optional behaviors that the user can opt into. Potentially something like:

1. (default behavior) throw on `null`s so the user knows they have to make some decision.
2. Turn `null` into `default`.
3. Convert nullable integer columns into `float`/`double`, and use `NaN` to designate `null` values. They can then use the Replace N/A transforms available to them in the rest of the pipeline.
4. The user can always change their schema (either inserting the data into a different table and replacing `null` as appropriate, creating a special stored proc or SELECT statement to do the `null` conversion) as an option to get around the exception as well.

See @TomFinley's comments at https://github.com/dotnet/machinelearning/issues/673#issuecomment-414886542 for more thoughts here.

@codemzs @ebarsoumMS - any thoughts on the appropriate behavior here?